base/samples/Transformation.hpp

```diff
@@ -0,0 +1,90 @@
+#ifndef BASE_SAMPLES_TRANSFORMATION_HPP
+#define BASE_SAMPLES_TRANSFORMATION_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a transformation in 3D of a local coordinate frame inside a global coordinate frame.
+ * A frame ID shall be globally unique and corresponds to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The transformation \f$ ^{global}_{local}T \f$ in this structure transforms a vector \f$ v_{local} \f$
+ * from the local into the global frame:
+ * \f$ v_{global} = ^{global}_{local}T \cdot v_{local} \f$
+ * In other words the transformation is the pose of the local frame expressed in the global frame.
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
+ */
+class Transformation
+{
+public:
+    /** Reference timestamp of the transformation sample */
+    base::Time time;
+
+    /** ID of the global reference frame */
+    std::string global_frame_id;
```

Comment:
This naming scheme is, to me, a LOT more confusing than RigidBodyState's convention (and that says something, given how I fought to understand RBS's convention). Which means that there's really no confusion (or, more likely, that confusion is unavoidable). What there is, is lack of documentation in RBS.

A transformation is not between a "global" and "local" frame. A 6D transformation /transforms/ a valid pose expressed in one frame into a valid pose expressed in another frame (that's the RBS convention)

Or a transformation is the pose of a frame into another (which is I guess what you call 'local' and 'global', but nothing's global here ... I have already a hard time explaining the fact that transformations are purely relative things to not have the types adding more to the confusion).

If you really want to go for the latter, then I would personally prefer

```
std::string frame_id;
std::string reference_frame_id;
```